src/Futhark/LSP/Utils.hs

```diff
@@ -0,0 +1,20 @@
+module Futhark.LSP.Utils
```

Comment:
This is not a good name for this module.  And how does it conceptually differ from the "Tools" module?

Reply:
I was thinking functions, etc defined in the `Utils` (utilities) module should be imported more than once.

While for `Tool` module, I put `getHoverInfoFromState` (only used once in `Handlers`) in it simply because it would be a bit crowded in `Handlers` otherwise.

What about I remove `Tool` and rename `Utils` to `Utilities`?

Reply:
I don't like the idea of "util" modules.  We have one already, `Futhark.Util`, but it is only for *truly* generic things.  The definition of the central state type for the LSP server is not really a "utility" - it is absolutely foundational.  Other parts of the compiler calls this kind of modules `Monad`, because they tend to define a monad used by other code.  That's not the case here.  Maybe `State` is a good name?  I imagine most of what will end up here will be state management.

(And actually, the `debug` function could be put in `Futhark.Util` as it is truly general.)

Reply:
Yep, that sounds like a better idea!